### ISSUE TITLE
bug: nvim-treesitter-yaml errors on large file with seemingly correct syntax

### ISSUE THREAD

**Issue by boydkelly** [tree-sitter-grammars/tree-sitter-yaml#9]
### Did you check existing issues?

- [X] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser
- [X] I have searched the existing issues

### Describe the bug

I have been working on this file with treesitter-yaml for a few years now with no issues.  However with the update to neovim .10 and changes to treesitter I am getting errors with my file.   This file is processed by yq and has no reported errors with lsp, yamlfmt or yamllint.   

File lexicon.yml located here: https://gitlab.com/ci-dict/dyu-xdxf

And used here: https://dyu-lex.coastsystems.net/dyu

### Steps To Reproduce/Bad Parse Tree

1. Open file lexicon.yaml with neovim 10 and up-to-date nvim-treesitter and yaml parser installed.
2. InspectTree will report an errror and highlighting will be wrong
3. Pin nvim-treesitter to 0.9.2 (or maybe 0.9.5)
4. Reinstall yaml query
5. Open file lexicon.yaml (No errro will be reported and highlighting will be correct)

### Expected Behavior/Parse Tree

InspectTree should not report errors as the file seems to be syntactically correct.

### Repro

https://gitlab.com/ci-dict/dyu-xdxf/-/blob/main/lexicon.yml?ref_type=heads

https://gitlab.com/ci-dict/dyu-xdxf.git

**Comment by boydkelly** [comment]
Yes it really seems this is just the file size.  If I truncated the file at line 32768 then treesitter will no longer report errors.

**Comment by ObserverOfTime** [comment]
Not much we can do about that.

**Comment by boydkelly** [comment]
Sorry, with all du respect for alll the hard word done here.  This used to work,  its a regression?   Could there not be some kind of performance option to fix this?

**Comment by ObserverOfTime** [comment]
It's not a tree-sitter-yaml issue.

**Comment by boydkelly** [comment]
You are so helpful.

**Comment by boydkelly** [comment]
Well Helix is using the older parser and it works ok.

**Comment by amaanq** [comment]
Hm, seems like https://github.com/tree-sitter-grammars/tree-sitter-yaml/commit/7b03feefd36b5f155465ca736c6304aca983b267 is the culprit, we're probably running out of space in the buffer...I don't think all of those casts have to be lossy

**Comment by ObserverOfTime** [comment]
`tree-sitter parse` works though.
